modules/BTBidAdapter.js

```diff
@@ -0,0 +1,184 @@
+const GVLID = 815;
+// reach out for approval to marketing@blockthrough.com
+const ENDPOINT_URL = 'https://pbs.sample.com';
+const SYNC_URL = 'https://cdn.sample.com';
+
+const CONVERTER = ortbConverter({
+  context: {
+    netRevenue: true,
+    ttl: 30,
+  },
+  imp,
+  request,
+  bidResponse,
+});
+
+/**
+ * Builds an impression object for the ORTB 2.5 request.
+ *
+ * @param {function} buildImp - The function for building an imp object.
+ * @param {Object} bidRequest - The bid request object.
+ * @param {Object} context - The context object.
+ * @returns {Object} The ORTB 2.5 imp object.
+ */
+function imp(buildImp, bidRequest, context) {
+  const imp = buildImp(bidRequest, context);
+  const { params, ortb2Imp } = bidRequest;
+
+  if (params) {
+    deepSetValue(imp, 'ext', params);
+  }
+  if (ortb2Imp?.ext.gpid) {
+    deepSetValue(imp, 'gpid', ortb2Imp.ext.gpid);
+  }
+
+  return imp;
+}
+
+/**
+ * Builds a request object for the ORTB 2.5 request.
+ *
+ * @param {function} buildRequest - The function for building a request object.
+ * @param {Array} imps - An array of ORTB 2.5 impression objects.
+ * @param {Object} bidderRequest - The bidder request object.
+ * @param {Object} context - The context object.
+ * @returns {Object} The ORTB 2.5 request object.
+ */
+function request(buildRequest, imps, bidderRequest, context) {
+  const request = buildRequest(imps, bidderRequest, context);
+  if (config.getConfig('debug')) {
+    request.test = 1;
+  }
+
+  return request;
+}
+
+/**
+ * Processes a bid response using the provided build function, bid, and context.
+ *
+ * @param {Function} buildBidResponse - The function to build the bid response.
+ * @param {Object} bid - The bid object to include in the bid response.
+ * @param {Object} context - The context object containing additional information.
+ * @returns {Object} - The processed bid response.
+ */
+function bidResponse(buildBidResponse, bid, context) {
+  const bidResponse = buildBidResponse(bid, context);
+  const { seat } = context.seatbid || {};
+  bidResponse.btBidderCode = seat;
+
+  return bidResponse;
+}
+
+/**
+ * Checks if a bid request is valid.
+ *
+ * @param {Object} bid - The bid request object.
+ * @returns {boolean} True if the bid request is valid, false otherwise.
+ */
+function isBidRequestValid(bid) {
+  if (!isPlainObject(bid.params) || !Object.keys(bid.params).length) {
+    logWarn('BT Bid Adapter: bid params must be provided.');
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * Builds the bid requests for the BT Service.
+ *
+ * @param {Array} validBidRequests - An array of valid bid request objects.
+ * @param {Object} bidderRequest - The bidder request object.
+ * @returns {Array} An array of BT Service bid requests.
+ */
+function buildRequests(validBidRequests, bidderRequest) {
+  const data = CONVERTER.toORTB({
+    bidRequests: validBidRequests,
+    bidderRequest,
+  });
+
+  return [
+    {
+      method: 'POST',
+      url: ENDPOINT_URL,
```

Comment:
you need a real url

Reply:
is that a requirement? I think we wanted to gate it behind approval.

Reply:
this doesnt appear to fall back to config; can you have a sensible default and fallback?